### ISSUE TITLE
Serial vs. parallel COPY FROM yield different node counts when importing from CSV file

### ISSUE THREAD

**Issue by prrao87** [kuzudb/kuzu#3504]
The number of lines in this CSV files as output by `wc -l` yields 14196 lines (excluding the header).

[postcode.csv](https://github.com/kuzudb/kuzu/files/15326739/postcode.csv)

However, when we read the CSV file into Kùzu using `COPY FROM` with parallelization disabled, the counts do not add up -- we obtain only 14190 nodes (6 nodes are missing). When parallelization is enabled in the CSV reader, the node count is correct, i.e., we get 14196 as expected.

### Serial copy

```py
import shutil
import kuzu

shutil.rmtree('./db', ignore_errors=True)
db = kuzu.Database('./db')
conn = kuzu.Connection(db, num_threads=1)

conn.execute("""
CREATE NODE TABLE PostCode(
    node_id STRING,
    code STRING,
    PRIMARY KEY (node_id))
""")

# import nodes
conn.execute('CALL THREADS=1')
conn.execute('COPY PostCode FROM "./data/postcode.csv" (HEADER=true, PARALLEL=false)')

queries = [
    'MATCH (n:PostCode) RETURN count(n)',
]

for query in queries:
    results = conn.execute(query)
    while results.has_next():
        print(results.get_next()[0])
```
Result (Incorrect):
```bash
$ python test.py
14190
```

### Parallel copy

```py
import shutil
import kuzu

shutil.rmtree('./db', ignore_errors=True)
db = kuzu.Database('./db')
conn = kuzu.Connection(db, num_threads=0)

conn.execute("""
CREATE NODE TABLE PostCode(
    node_id STRING,
    code STRING,
    PRIMARY KEY (node_id))
""")

# import nodes
conn.execute('COPY PostCode FROM "./data/postcode.csv" (HEADER=true)')

queries = [
    'MATCH (n:PostCode) RETURN count(n)',
]

for query in queries:
    results = conn.execute(query)
    while results.has_next():
        print(results.get_next()[0])
```
Result (correct):
```bash
$ python test.py
14196
```

**Comment by prrao87** [comment]
If it helps narrow things down, I've done a diff of which postcodes are absent in the serial copy case:
```
['WN2 2EG', 'BL8 1PD', 'M6 6QT', 'SK15 1EL', 'OL8 3XA', 'M19 3HL']
```

**Comment by acquamarin** [comment]
This bug has been solved in #3505.
